src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2229,17 +2226,20 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
 
             var receiver = node.Receiver;
             var receiverType = VisitRvalueWithState(receiver);
+            _currentConditionalReceiverVisitResult = _visitResult;
```

Comment:
Consider a `Debug.Assert(_currentConditionalReceiverVisitResult == default` predicate to this method.

Reply:
I think that wouldn't be the case in `x?.y?.f` when we get the `y`. I'll double-check.